packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts

```diff
@@ -48,397 +28,167 @@ const supportedCurrencies: Record<string,  Record<RequestLogicTypes.CURRENCY, st
+      throw Error('acceptedTokens is required');
+    }
+    if (
+      creationParameters.acceptedTokens.some((address: string) => !this.isValidAddress(address))
+    ) {
+      throw Error('acceptedTokens must contains only valid ethereum addresses');
     }
 
-    copiedExtensionState[extensionAction.id] = applyCreation(extensionAction, timestamp);
-
-    return copiedExtensionState;
-  }
-
-  // if the action is not "create", the state must have been created before
-  if (!requestState.extensions[extensionAction.id]) {
-    throw Error(`The extension should be created before receiving any other action`);
-  }
-
-  if (extensionAction.action === ExtensionTypes.PnFeeReferenceBased.ACTION.ADD_PAYMENT_ADDRESS) {
-    copiedExtensionState[extensionAction.id] = ReferenceBased.applyAddPaymentAddress(
-      isValidAddress,
-      copiedExtensionState[extensionAction.id],
-      extensionAction,
-      requestState,
-      actionSigner,
-      timestamp,
-    );
-
-    return copiedExtensionState;
-  }
-
-  if (extensionAction.action === ExtensionTypes.PnFeeReferenceBased.ACTION.ADD_REFUND_ADDRESS) {
-    copiedExtensionState[extensionAction.id] = ReferenceBased.applyAddRefundAddress(
-      isValidAddress,
-      copiedExtensionState[extensionAction.id],
-      extensionAction,
-      requestState,
-      actionSigner,
-      timestamp,
-    );
-
-    return copiedExtensionState;
-  }
+    const network = creationParameters.network;
+    if (!network) {
+      throw Error('network is required');
+    }
+    if (!supportedCurrencies[network]) {
+      throw Error('network not supported');
+    }
 
-  if (extensionAction.action === ExtensionTypes.PnFeeReferenceBased.ACTION.ADD_FEE) {
-    copiedExtensionState[extensionAction.id] = applyAddFee(
-      copiedExtensionState[extensionAction.id],
+    const supportedErc20: string[] = supportedCurrencies[network][RequestLogicTypes.CURRENCY.ERC20];
+    if (
+      creationParameters.acceptedTokens.some((address: string) => !supportedErc20.includes(address))
+    ) {
+      throw Error('acceptedTokens must contain only supported token addresses (ERC20 only)');
+    }
+    return super.createCreationAction(creationParameters);
+  }
+
+  /**
+   * Applies the extension action to the request state
+   * Is called to interpret the extensions data when applying the transaction
+   *
+   * @param extensionsState previous state of the extensions
+   * @param extensionAction action to apply
+   * @param requestState request state read-only
+   * @param actionSigner identity of the signer
+   *
+   * @returns state of the request updated
+   */
+  public applyActionToExtension(
+    extensionsState: RequestLogicTypes.IExtensionStates,
+    extensionAction: ExtensionTypes.IAction,
+    requestState: RequestLogicTypes.IRequest,
+    actionSigner: IdentityTypes.IIdentity,
+    timestamp: number,
+  ): RequestLogicTypes.IExtensionStates {
+    return super.applyActionToExtension(
```

Comment:
if the only thing a child method does is call its parent, then maybe it can be removed?